lib/segment/src/index/hnsw_index/graph_links.rs

```diff
@@ -224,9 +255,10 @@ impl GraphLinksConverter {
             let links_range = header.get_links_range();
             let offsets_range = header.get_offsets_range();
             let union_range = links_range.start..offsets_range.end;
-            let (links_mmap, offsets_mmap) = bytes_data[union_range]
+            let (links_mmap, offsets_with_padding_mmap) = bytes_data[union_range]
                 .as_mut()
                 .split_at_mut(links_range.len());
+            let offsets_mmap = &mut offsets_with_padding_mmap[header.offsets_padding as _..];
```

Comment:
I've tweaked @IvanPleshkov's code here, to make sure `offsets_padding` betes *aren't* included as (an invalid) last element of `links_mmap`.

But, please, check my slicing here....

Reply:
> I've tweaked @IvanPleshkov's code here, to make sure `offsets_padding` betes _aren't_ included as (an invalid) last element of `links_mmap`. (See [this comment](https://github.com/qdrant/qdrant/pull/3807#discussion_r1521806532) from #3807.)
> 
> But, please, check my slicing here....

Thanks for the fix, you are totally right and I like your opinion!